consul/watcher.go

```diff
@@ -153,6 +162,10 @@ func (w *Watcher) startUpstream(up api.Upstream) {
 		Datacenter:       up.Datacenter,
 	}
 
+	if up.Config["protocol"] != nil {
+		u.Protocol = up.Config["protocol"].(string)
```

Comment:
There should be a type assertion here. If `"protocol"` is not a string it will crash.

Reply:
Thanks for catching this. I added a type assertion